### PR TITLE
Test: Fix skipped failing integration tests

### PR DESCRIPTION
**Pull request by rackstar**
## Context

Fix failing integration tests due to dynamic pricing

(Might be easier to review commit by commit)

## Changes proposed in this pull request

* [x] fix getInternalTokenPriceInAsset integration tests
* [x] fix getTotalActiveCoverAmount integration tests
* [x] fix `InsufficientCapacity` error in recalculate effective weights integration tests
* [x] fix skipped expired cover integration test 

## Test plan

Removed the .skip and ensure the previous failing tests are now passing

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [x] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: test/integration/Cover/expireCover.js [hunk 1]
```diff
@@ -242,12 +238,6 @@ describe('expireCover', function () {
         { value: amount },
       );
 
-    const allocationsAfter = await stakingPool1.getActiveAllocations(productId);
-    const { totalActiveCoverInAsset: activeCoverAfter } = await cover.activeCover(coverAsset);
-
-    // stays the same because same amount expired and was allocated
-    expect(sum(allocationsBefore)).to.be.equal(sum(allocationsAfter));
-    expect(activeCoverBefore).to.be.equal(activeCoverAfter);
```
Comment (rackstar) [inline test/integration/Cover/expireCover.js:250]: I'm not 100% certain but it seems its possible that these expectations no longer hold true due to the dynamic pricing.

The cover amount from the first and second `buyCover` is the same but the token price is different.

File: test/integration/MCR/getTotalActiveCoverAmount.js [hunk 2]
```diff
@@ -67,103 +83,101 @@ describe('getTotalActiveCoverAmount', function () {
...
+      .add(expectedEthCoverAmount2)
+      .add(expectedDaiCoverAmount1)
+      .add(expectedDaiCoverAmount2);
+    const actualTotalActiveCoverAmount = await mcr.getTotalActiveCoverAmount();
+
+    const totalActiveCoverAmountDiff = expectedTotalActiveCoverAmount - actualTotalActiveCoverAmount;
+    expect(
+      totalActiveCoverAmountDiff,
+      `Total active cover amount ${actualTotalActiveCoverAmount} not close enough to ${expectedTotalActiveCoverAmount}`,
+    ).to.be.lessThanOrEqual(1); // <= 1 wei
```
Comment (rackstar) [inline test/integration/MCR/getTotalActiveCoverAmount.js:181]: When adding up all the multiple purchase covers it has a discrepancy of **1 wei** from the the expect amount. This seems to be an acceptable amount hence the tests is written this way.

(single purchase covers does not have this discrepancy)

File: test/integration/Pool/pool.js [hunk 3]
```diff
@@ -49,50 +50,46 @@ async function tokenPriceSetup() {
...
+
     const daiBalance = await dai.balanceOf(pool.address);
-    const expectedDAiValueInEth = daiToEthRate.mul(daiBalance).div(parseEther('1'));
-    const expectedTotalAssetValue = poolBalance.add(expectedDAiValueInEth);
-    const totalAssetValue = await pool.getPoolValueInEth();
-    assert(totalAssetValue.toString(), expectedTotalAssetValue.toString());
+    const expectedDaiValueInEth = daiToEthRate.mul(daiBalance).div(parseEther('1'));
+
+    const usdcBalance = await usdc.balanceOf(pool.address);
+    const expectedUsdcValueInEth = usdcToEthRate.mul(usdcBalance).div(parseUnits('1', USDC_DECIMALS));
```
Comment (rackstar) [inline test/integration/Pool/pool.js:83]: The assertion for this test was previously wrong so it was always passing even though it was not taking into account the USDC balance.

```javascript
assert(totalAssetValue.toString(), expectedTotalAssetValue.toString()); // wrong assertion
```

File: test/integration/Pool/pool.js [hunk 4]
```diff
@@ -49,50 +50,46 @@ async function tokenPriceSetup() {
...
+
+    const totalAssetValue = await pool.getPoolValueInEth();
 
     const poolBalance = BigNumber.from(await ethers.provider.getBalance(pool.address));
+
     const daiBalance = await dai.balanceOf(pool.address);
-    const expectedDAiValueInEth = daiToEthRate.mul(daiBalance).div(parseEther('1'));
-    const expectedTotalAssetValue = poolBalance.add(expectedDAiValueInEth);
-    const totalAssetValue = await pool.getPoolValueInEth();
-    assert(totalAssetValue.toString(), expectedTotalAssetValue.toString());
```
Comment (rackstar) [inline test/integration/Pool/pool.js:89]: This assertion is wrong so the test was always passing even though it was not taking into account the USDC balance

File: test/unit/Ramm/setup.js [hunk 5]
```diff
@@ -10,13 +10,6 @@ const { parseEther } = ethers.utils;
 const SPOT_PRICE_A = parseEther('0.0347');
 const SPOT_PRICE_B = parseEther('0.0152');
 
-async function getState(ramm) {
-  const { nxmReserveA: nxmA, nxmReserveB: nxmB } = await ramm.slot0();
-  const { ethReserve: eth, budget, updatedAt: timestamp } = await ramm.slot1();
-  const ratchetSpeed = await ramm.ratchetSpeed();
-  return { nxmA, nxmB, eth, budget, ratchetSpeed, timestamp };
-}
-
```
Comment (rackstar) [inline test/unit/Ramm/setup.js:19]: This helper function was written before we wrote `ramm.loadState`. Removing in favour of `loadState` for consistency.

**Comment by shark0der** [comment]
Closed in favor of #1019

**Pull request closed without merging**